obspy/io/sac/tests/test_core.py

```diff
@@ -804,6 +804,27 @@ def test_valid_sac_from_minimal_existing_sac_header(self):
         self.assertEqual(tr1.stats.sac.stla, 1.0)
         self.assertEqual(tr1.stats.sac.stlo, 2.0)
 
+    def test_merge_sac_obspy_headers(self):
+        """
+        Test that manually setting a set of SAC headers not related
+        to validity or reference time on Trace.stats.sac is properly merged
+        with the Trace.stats header. Issue 1285.
+        """
+        tr = Trace(data=np.arange(30))
+        o = 10.0
+        tr.stats.sac = {'o': o}
+
+        with NamedTemporaryFile() as tf:
+            tempfile = tf.name
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter('always')
+                tr.write(tempfile, format='SAC')
```

Comment:
Reusing the file without closing it is usually troublesome on Windows; I'm surprised AppVeyor was okay with it.

Is there no reason you can't just use the `file` object `tf` directly?